### ISSUE TITLE
FNX2-12995 ⁃ Add new Leanplum E_Changed_Default_To_Fenix Event

### ISSUE THREAD

**Issue by cbonacuse** [mozilla-mobile/fenix#12103]
The following were identified as missing in Fenix during a recent test by the Mobile Engagement team. This event by a similar name (E_Changed_Default_To_Fennec) is present in Fennec. This is requested to be in market by 8/25/20 (release 80).

* [ ] E_Changed_Default_To_Fenix

This should be fired when the user changes their device setting to have the default browser sent to Firefox (Fenix), regardless of the app being presently opened or not.

Tasks:

* [ ]  Add mapping for E_Changed_Default_To_Fenix
* [ ]  Undergo data review
* [ ]  Add test to show it's being fired (e.g. specified by #11210 )
* [ ] Test end-to-end and validate attribute in Leanplum

**Comment by ekager** [comment]
cc @cbonacuse this has landed!

**Comment by AndiAJ** [comment]
Hi @cbonacuse & @sblatz , I've just checked this matter on both the latest **`Nightly Build 200730`** and **`Fenix Beta 80.0.0-beta-1`** using the following devices:
• Google Pixel 3a (Android 10)
• Huawei Mate 20 Lite (Android 9)
• OnePlus A3 (Android 6.0.1)

The `E_Changed_Default_To_Fenix` event **is properly generated**, on the other hand it looks like the `default_browser` attribute remains empty in most cases, filled #13122


## ✔️ Nightly
Google Pixel 3a -  device ID **`e6222324-6a2d-403a-abc5-a7b777227632`**
![image](https://user-images.githubusercontent.com/51314259/88920724-ffa17780-d275-11ea-8d80-af091ca8ec76.png)
![image](https://user-images.githubusercontent.com/51314259/88920685-ed273e00-d275-11ea-8d9e-8798ed3ad1df.png)

Huawei Mate 20 Lite -  device ID **`1334424f-5898-4337-9ed7-9679fed9fb25`**
![image](https://user-images.githubusercontent.com/51314259/88920819-265fae00-d276-11ea-9726-a1c9addd3497.png)
![image](https://user-images.githubusercontent.com/51314259/88920907-4d1de480-d276-11ea-85ba-5af43fb4eea0.png)

OnePlus A3 -  device ID **`d5e35e4f-c75a-4b5e-9aa0-6aa2cbb6c78b`**
![image](https://user-images.githubusercontent.com/51314259/88920998-73dc1b00-d276-11ea-88e4-7c1756ecf4a5.png)
![image](https://user-images.githubusercontent.com/51314259/88921027-7c345600-d276-11ea-8235-e6fa5897c97c.png)



## ✔️ Beta
Google Pixel 3a - device ID **`35d5c40d-0362-4b55-8aee-ffc6a9e7012c`**
![image](https://user-images.githubusercontent.com/51314259/88917406-5efc8900-d270-11ea-92e9-4bc61532eb41.png)
![image](https://user-images.githubusercontent.com/51314259/88917439-6ae84b00-d270-11ea-8b0d-048e58292b10.png)

Huawei Mate 20 Lite - device ID 48e981ae-5ec4-43f7-bbb2-5d4532f15627
![image](https://user-images.githubusercontent.com/51314259/88917504-894e4680-d270-11ea-851e-d81a66aab958.png)
![image](https://user-images.githubusercontent.com/51314259/88917535-95d29f00-d270-11ea-94a1-222a6f962480.png)


OnePlus A3 - device ID **`27951d4c-5729-43cf-92e2-ad7553a1a5c5`**
![image](https://user-images.githubusercontent.com/51314259/88917591-b1d64080-d270-11ea-997b-a3f78825ca73.png)
![image](https://user-images.githubusercontent.com/51314259/88917618-bf8bc600-d270-11ea-9a06-92425093182d.png)

**Comment by cbonacuse** [comment]
@AndiAJ - it looks like the Beta screenshots are for the Search Widget event - did you try with this E_Changed_Default_To_Fenix one?

@jhugman - I tested this on my Google Pixel 3 with Nightly Build 200730 and am not seeing the event fire, even though the default_browser attribute is correctly changing from org.mozilla.firefox to org.mozilla.fenix. I tried both cold starting and also backgrounding the app, opening the app, and then killing.

**Comment by AndiAJ** [comment]
Hi @cbonacuse , my bad, mixed up the screenshots, sorry for the inconvenience. ☺️ 
Re-checked it today and the **`E_Changed_Default_To_Fenix`** is properly generated.

## ✔️ Beta
Google Pixel 3a - device ID **`3d68a448-f183-4f0e-bc6e-6a3049c85d6d`**
![image](https://user-images.githubusercontent.com/51314259/89001256-61a7be80-d302-11ea-8c07-d9fa6ab62726.png)

Huawei Mate 20 Lite - device ID **`2d0fe138-a652-4d72-81e3-62a60b1a7017`**
![image](https://user-images.githubusercontent.com/51314259/89001199-2dcc9900-d302-11ea-824d-2f0f702d4d17.png)

OnePlus A3 - device ID **`a3f8e20f-cefa-46bd-967a-c5dd963e367b`**
![image](https://user-images.githubusercontent.com/51314259/89001229-45a41d00-d302-11ea-87fc-cc75dda787fd.png)

**Comment by sblatz** [comment]
@cbonacuse are you still not seeing it on your end? Want to make sure this is properly resolved 😄

**Comment by cbonacuse** [comment]
@sblatz thanks for following up! I am not still :(

**Comment by sblatz** [comment]
@AndiAJ are you able to tell me the steps you used to verify this so @cbonacuse can confirm on her end?

Also are you verifying using Glean or Amplitude 🤔

**Comment by AndiAJ** [comment]
Hi @sblatz & @cbonacuse , I'm using only Leanplum.

1. Identify the devices I use and register them as test devices.
2. Open each test device in a separate tab.
3. After setting for E.g. Nightly as the default browser I open a page (or perform any other action) and put Fenix in background.
4. Refresh the tabs from Step 2
5. Check the Timeline section
  
**► Re-checked on today's Nightly Build** 

Google Pixel 3a - device ID `f5921b8d-314f-40f4-aa19-97360d5d94b8`
![image](https://user-images.githubusercontent.com/51314259/89302081-14fe1380-d673-11ea-91e9-ee5308af7b57.png)

Huawei Mate 20 Lite - device ID `af0c049d-bed4-4c99-a917-e03e7fd42d2a`
![image](https://user-images.githubusercontent.com/51314259/89301962-e7b16580-d672-11ea-8e10-f329a1f560be.png)

OnePlus A3 - device ID `74f1b775-d6e0-40e9-a804-0d374398b7a5`
![image](https://user-images.githubusercontent.com/51314259/89301998-f9930880-d672-11ea-839c-39c1350beeb2.png)

**Comment by cbonacuse** [comment]
@AndiAJ & @sblatz I just did another test myself with the following steps and I'm still not seeing this on my Google Pixel 3 (OS 10 & Nightly build 200805) when I do a cold start or background the app. I'm checking in Leanplum as well.

My Leanplum ID: 4ead7db7-0a20-4f5e-911b-4097b3db4048

When you say 'open device in a separate tab', are you not using an actual device but rather simulating it somehow?

1. Check device settings - Chrome is default browser
2. Check attribute -  `default_browser = null`
3. Go to device settings, set Nightly as default browser
4. Open Nightly
5. Check attribute - `default_browser = org.mozilla.fenix`
6. Check events - no event fired except E_Opened_App
7. Kill app
8. Check events - no event fired 
9. Open app again 
10. Background app
11. Open app from background
12. Kill app

**Comment by AndiAJ** [comment]
Hi @cbonacuse & @sblatz I've just re-checked this matter on Nightly 200805 with a Google Pixel 3a (Android 10) using the above mentioned STR [**added a couple of notes and an extra step (Step 6)]** and everything seems to be working fine.

Device ID - `d853ec8d-3230-45a9-be08-0e147b033257`

1. Check device settings - Chrome is default browser
2. Check attribute - default_browser = null
3. Go to device settings, set Nightly as default browser
4. Open Nightly
5. Check attribute - default_browser = null (❗ **it will be updated only after close + resume**)
6. Force close and resume
7. Re-check attribute - ✔️ **default_browser = org.mozilla.fenix**
8. Check events - ✔️ **E_Changed_Default_To_Fenix event fired**
![image](https://user-images.githubusercontent.com/51314259/89497225-9913e080-d7c4-11ea-98d1-583442c87081.png)
![image](https://user-images.githubusercontent.com/51314259/89497266-ab8e1a00-d7c4-11ea-89f5-1386aaa44af7.png)

**Comment by cbonacuse** [comment]
@AndiAJ I followed your same steps above and I'm still not seeing the event for 4ead7db7-0a20-4f5e-911b-4097b3db4048. Could it have something to do with my device (Google Pixel 3 - yours is a 3a)?

**Comment by AndiAJ** [comment]
Hi, @cbonacuse , yes, Google Pixel **3a**
Not sure why you aren't seeing the event fired, as you can see in the previous comments it worked properly on other devices (Huawei Mate 20 Lite (Android 9), OnePlus A3 (Android 6.0.1))

I've checked the timeline of your device ID and it dates back to the 3rd of April.
Did you try with a freshly installed Nightly? maybe this will do the trick. ☺️

**Comment by cbonacuse** [comment]
@AndiAJ I can definitely try to install it again and see if that works. Is this in Beta yet as well? I can also test there.

**Comment by AndiAJ** [comment]
Hi @cbonacuse , yes it is available on today's [Beta build 80.0.0-beta.4. (task index link if you want to install it)](https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.fennec-beta.latest)
Just checked it with a OnePlus A3 (Android 6.0.1) and it works.
![image](https://user-images.githubusercontent.com/51314259/89657202-5b52ad00-d8d5-11ea-87a3-0d403a69c61e.png)

**Comment by cbonacuse** [comment]
@AndiAJ I deleted and then re-installed Nightly and found my new device ID. I'm still not having the event fire for me but I see the attribute is changing showing that Nightly is my default browser.

Device ID: 53d503bf-1812-46c0-ab26-fa42f7e00bb9
OS: 10
Device: Google Pixel 3
App version: 200805 build (I downloaded on 8/7 - has there not been an update since 8/5?)

Steps to reproduce:

1. Uninstall Nightly
2. Re-install Nightly
3. Perform some actions to find my device in Leanplum (ex: add search widget and clear private data)
4. Change device setting to set Nightly as default browser
5. Open Nightly
6. Kill Nightly
7. Check attribute - `default_browser = org.mozilla.fenix`
8. Check event - only `E_Opened_App` fired

Happy to hop on a call next week to discuss this further too @sblatz
